ada_2_fhir/zibs2017/payload/_zib2017.xsl

```diff
@@ -27,8 +27,7 @@ The full text of the license is available at http://www.gnu.org/copyleft/lesser.
     <xsl:param name="macAddress">02-00-00-00-00-00</xsl:param>
 
     <xsl:variable name="patients" as="element()*">
-        <!-- Patiënten -->
-        <xsl:for-each-group select="//patient[not(patient)][not(@datatype = 'reference')][.//(@value | @code | @nullFlavor)]" group-by="
+        <xsl:for-each-group select="//patient[not(patient)][ancestor::subject][not(@datatype = 'reference')][.//(@value | @code | @nullFlavor)]" group-by="
```

Comment:
I don't know for certain, to be honest. I've added `[ancestor::subject]` to make sure the following part of the ada instances does not result in 'empty' Patient fixtures:

<information_source conceptId="2.16.840.1.113883.2.4.3.11.60.40.1.0.0.2">
  <patient_as_information_source conceptId="2.16.840.1.113883.2.4.3.11.60.40.1.0.0.3">
    <patient conceptId="2.16.840.1.113883.2.4.3.11.60.7.2.140" nullFlavor="NI"/>
  </patient_as_information_source>
</information_source>

Would you suggest just removing these `information_source`s from the ada instances?

Reply:
I don't believe all dataset will have 'subject' as parent, so that would fail. If we need a proper patient (that is not empty ánd not a reference) than I would suggest to check for [*//(@value | @code | @nullFlavor)] instead of [.//(@value | @code | @nullFlavor)], because indeed now this nullFlavor patient will appear, and we don't want that. Alternative is to remove the nullFlavor, because what good is a patient with subelements where none of them have an @value or @code .

Reply:
Changed `select` to your suggestion.